js/activity-works.js

```diff
@@ -286,4 +286,4 @@ define(function (require) {
             update = true;
         }
     });
-});
+});
```

Comment:
I don't know how you keep removing the new line at EOF, but you can add it with `echo >> filename`.

Reply:
sure @chimosky i will use echo form now on

Reply:
Using echo is fixing it after the event.  A useful habit is to
* have a view of `git diff` open so you can see what you have actually changed; with that you should have spotted the removal of newline,
* use an interactive `git add` so you can _approve_ what to add to each commit; with that you should have avoided inserting the removal of newline into a commit, and we wouldn't have to see it.

There are software tools and plugins that provide these techniques.  I use Emacs and Magit to do it, as well as a Terminal.  I've heard others using VScode, but I don't know the plugin names.